model/modelprocessor/datastream.go

```diff
@@ -141,10 +141,18 @@ func metricsetDataset(event *modelpb.APMEvent) string {
 		// metrics that we don't already know about; otherwise they will end
 		// up creating service-specific data streams.
 		internal := true
-		for _, s := range event.Metricset.Samples {
-			if !IsInternalMetricName(s.Name) {
-				internal = false
-				break
+
+		// set internal to false for metrics translated using OTel remappers.
+		if label, ok := event.Labels["event.provider"]; ok && label != nil {
+			internal = !(label.Value == "hostmetrics")
+		}
+
+		if internal {
+			for _, s := range event.Metricset.Samples {
+				if !IsInternalMetricName(s.Name) {
+					internal = false
+					break
+				}
```

Comment:
[**For reviewers**] This code is required to make sure that the remapped OTel metrics end up in the app datastream instead of the internal datastream.